### PR TITLE
Updates fileSize limits example in README

### LINKED ISSUE #250: limits.fileSize not working when greater than a specified value
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported

### Fastify version

3.8.1

### Plugin version

4.0.7

### Node.js version

14.17.1

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

Manjaro 21

### Description

When fileSize is greater than a specified value, it will not throw exception when upload large file.

### Steps to Reproduce

Here is the sample code just from README

```js
const fastify = require('fastify')()
const fs = require('fs')
const util = require('util')
const path = require('path')
const { pipeline } = require('stream')
const pump = util.promisify(pipeline)

fastify.register(require('fastify-multipart'))

fastify.post('/', async function (req, reply) {
  // process a single file
  // also, consider that if you allow to upload multiple files
  // you must consume all files otherwise the promise will never fulfill
  console.log('before req file');
  // {limits: {fileSize: 1024}}
  const data = await req.file({limits: {fileSize: 40 * 1024}})
  console.log(data);
  console.log('after req file');

  // to accumulate the file in memory! Be careful!
  //
  // await data.toBuffer() // Buffer
  //
  // or

  console.log('before pump')
  await pump(data.file, fs.createWriteStream(data.filename))
  console.log('after pump')
  // be careful of permission issues on disk and not overwrite
  // sensitive files that could cause security risks
  
  // also, consider that if the file stream is not consumed, the promise will never fulfill

  reply.send()
})

fastify.listen(3000, err => {
  if (err) throw err
  console.log(`server listening on ${fastify.server.address().port}`)
})
```

Test with `curl`:

```sh
curl -v -F 'file=@BIG_FILE' localhost:3000/
```

When `fileSize` is a small one like `40 * 1024`(40KB), it's working, response:

```json
{"statusCode":413,"code":"FST_REQ_FILE_TOO_LARGE","error":"Payload Too Large","message":"request file too large, please check multipart config"}
```

But when `fileSize` is a large one like `40 * 1024 * 1024`(40MB), it's not working, the file is uploaded successfully, but the uploaded file is truncated to 40MB only.

### Expected Behavior

`limits.fileSize` should working as expect.

### LINKED ISSUE #256: limits.fileSize when set global fileSize doesn't get limited
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported

### Fastify version

3.0.0

### Plugin version

4.0.5

### Node.js version

14.17.3

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

20.04

### Description

I'm setting up the plugin to limit to file size of 2 MB, it doesn't work this is my current setup
image

### Steps to Reproduce

As documented here `limit.fileSize` didn't work for me when set global(On plugin registration)

### Expected Behavior

The plugin should have limited to the size I've specified during its registration

### PR DESCRIPTION
**Pull request by leorossi**
Updates the example to work with `limits.fileSize` option.

Closes #256, Closes #250 

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by simoneb — APPROVED** [review]

**Review by simoneb — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -90,21 +90,18 @@ fastify.register(require('fastify-multipart'), {
 });
 ```
 
-If you do set upload limits, be sure to catch the error. An error or exception will occur if a limit is reached. These events are documented in more detail [here](https://github.com/mscdex/busboy#busboy-special-events).
```
Comment (mcollina) [inline README.md:93]: why was this removed?

File: README.md [hunk 2]
```diff
@@ -90,21 +90,18 @@ fastify.register(require('fastify-multipart'), {
...
 
 ```js
-try {
-  const data = await req.file()
-  await pump(data.file, fs.createWriteStream(data.filename))
-} catch (error) {
-  if (error instanceof fastify.multipartErrors.FilesLimitError) {
-    // handle error
-  }
+const data = await req.file()
```
Comment (StarpTech) [inline README.md:99]: LGTM. We should try to promote `const files = await req.saveRequestFiles()` higher than the lowe level api. As you already experience there are some edge cases. `saveRequestFiles()` is good for most of the use cases.

File: README.md (hunk 1 above)
Comment (leorossi) [reply README.md:93]: Because the error is not thrown actually, check this comment: https://github.com/fastify/fastify-multipart/issues/250#issuecomment-888356293

**Comment by simoneb** [comment]
@mcollina @StarpTech are we happy with this change? If so, can you please merge this PR?

**Review by mcollina — APPROVED** [review]
lgtm
